include/nanorange/algorithm/stable_partition.hpp

```diff
@@ -95,37 +98,40 @@ struct stable_partition_fn {
         }
 
         const I first_false = (m1 == first) ? first :
-                impl_unbuffered(first, m1, len_half, pred, proj);
+                impl_unbuffered(first, m1, len_half, pred, proj).end();
 
         m1 = middle;
         len_half = dist - half;
 
         while (nano::invoke(pred, nano::invoke(proj, *m1))) {
             if (++m1 == last) {
-                return nano::rotate(first_false, middle, ++last).begin();
+                auto rot = nano::rotate(first_false, middle, ++last);
+                return {rot.begin(), rot.end()};
```

Comment:
This looks wrong. If it used to return `rot.begin()`, then that should be the sentinel for the returned `subrange<I>`. This should probably return `{old_first, rot.begin()}`